Snowplow/Internal/Tracker/SPServiceProviderProtocol.h

```diff
@@ -25,7 +25,7 @@
 @class SPSubject;
 @class SPEmitter;
 @class SPTracker;
-@class SPTrackerControllerImpl;
+@protocol SPTrackerController;
 @class SPEmitterControllerImpl;
 @class SPNetworkControllerImpl;
 @class SPGDPRControllerImpl;
```

Comment:
Is the goal to eventually change all these to protocols and `id<T>`?

Reply:
There isn't a clear goal, honestly.
I needed to change it because passing the implementation of the protocol (SPTrackerControllerImpl instead of id<SPTrackerController>) raised a warning.

In term of the goal for ServiceProviderProtocol, I think it would be useful to improve testing, so yes, passing the protocols can be a solution but I'll take the decision when it will be needed for testing (or other requirements).
I haven't thought in the deep about this yet.